### DISCUSSION TITLE
Allowed memory size exceeded using findAll()

### DISCUSSION THREAD

**Discussion by rookiextreme** [symfony/symfony#57711] [Q&A]
Hi, new to symfony.

so the entitiy affected right now is:

1-M 
Country -> State

M-1 
State -> Country

When using

$a = CountryRepository::findAll()
print_r($a)

Got an error
Allowed memory size exceeded

If state relation property in country is deleted, it will show no error and views all country

Eventhough with allowed memory size exceeded,
The count of country is correct

Followed the documentation but error still have

any help would be appreciated

**Comment by alexandre-le-borgne** [comment]
Hello,

At the very least, add the code for your entities and the complete error trace to hope to get a relevant response.

**Reply by rookiextreme** [reply]
hi, sorry for late response.

Here is the code:
PropertiesController.php
```
public function index(String $type, PropertyRepository $propertyRepository, CountryRepository $countryRepository, StateRepository $stateRepository, Integer $id = null): Response
    {
        $property = !$id ? new Property() : $propertyRepository->find($id);
        $country = $countryRepository->findAll();
        echo '<pre>';
        print_r($country);
        echo '</pre>';
        die();
   }
```

Country.php(entity)
```
public function __construct()
    {
        $this->states = new ArrayCollection();
    }
#[ORM\OneToMany(targetEntity: State::class,mappedBy: 'country', cascade: ['persist', 'remove'])]
    private Collection $states;

public function getStates(): Collection
    {
        return $this->states;
    }
```

State.php(entity)
```
 #[ORM\ManyToOne(inversedBy: 'state')]
    private ?Country $country = null;
```
errors are below, i just wana see the data before any loops basically
<img width="1231" alt="image" src="https://github.com/user-attachments/assets/cf987eae-8769-45c2-9d2a-04174c1789fa">

**Reply by rookiextreme** [reply]
And a question, always used to using print_r for showing the objects of a query, just found that if using function dump($country) its fine..but with print_r(allowed memory size issue)..is there a difference?

**Reply by stof** [reply]
`print_r` is very dumb in its processing, which does not work well with circular object graphs (which is the case here with your bidirectional relation).
Thus, when you have a non-initialized proxy in the dumped data, `print_r` would end up including the whole internal state of the ORM (as the proxy has a reference to it in its initializer)

The VarDumper component has specific handling for those cases.

**Reply by rookiextreme** [reply]
So its actually applies to entities with relation am i right? Because its in a loop because entity A link to Entity B and B link to A?

meaning by using findAll, it queries even the relations? What if query just Entity A with no relation? Sorry if dumb question..

**Reply by stof** [reply]
if an entity has no relation, it is much less likely to involve a circular object graph.

But anyway, VarDumper is a much better debugging tool than `print_r`

**Reply by rookiextreme** [reply]
Understood….but how does laravel deal with this? Because print_r seems to work fine there with the all() method which is equivalent to findAll()

Just want to understand it because its new knowledge
